workflows/wf3/2-map-metadata.js

```diff
@@ -0,0 +1,62 @@
+const isValidValue = value => value !== '' && value !== 'NA';
+
+fn(state => {
+  const { f01MphssBaselineXls } = state;
+  const keys = f01MphssBaselineXls[1];
+
+  const mphssBaselineXls = f01MphssBaselineXls.slice(2).map(item => {
+    const obj = item.reduce((acc, value, idx) => {
+      acc[keys[idx]] = value;
+      return acc;
+    }, {});
+
+    return obj;
+  });
+
+  state.f01MphssBaseline = mphssBaselineXls
+    .filter(
+      o =>
+        isValidValue(o['External ID']) && isValidValue(o['DHIS2 DE full name'])
+    )
+    .map(o => {
+      return {
+        'value.display - Answers': o['Answers'],
+        'value.uuid - External ID': o['External ID'],
+        'DHIS2 DE full name': o['DHIS2 DE full name'],
+        'DHIS2 DE UID': o['DHIS2 DE UID'],
+        'OptionSet name': o['OptionSet name'],
+        'DHIS2 Option Set UID': o['DHIS2 Option Set name'],
+        'DHIS2 Option name': o['DHIS2 Option name'],
+        'DHIS2 Option UID': o['DHIS2 Option UID'],
+        'DHIS2 Option Code': o['DHIS2 Option code'],
+      };
+    });
+
+  return state;
+});
+
+fn(state => {
+  const { optionSetsXlsData } = state;
+  const keys = optionSetsXlsData[1];
+
+  const optsMap = optionSetsXlsData.slice(2).map(item => {
```

Comment:
Is this function the same for each block? If so you should declare it as a const at the top of the file. It'll be more readable too